arch/x86/src/boundary/boundary_impl.rs

```diff
@@ -0,0 +1,254 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2024.
+
+use core::fmt::Write;
+
+use crate::registers::bits32::eflags::{EFlags, EFLAGS};
+
+use kernel::process::FunctionCall;
+use kernel::syscall::{ContextSwitchReason, Syscall, SyscallReturn, UserspaceKernelBoundary};
+use kernel::ErrorCode;
+
+use crate::interrupts::{IDT_RESERVED_EXCEPTIONS, SYSCALL_VECTOR};
+use crate::segmentation::{USER_CODE, USER_DATA};
+
+use super::UserContext;
+
+/// Defines the usermode-kernelmode ABI for x86 platforms.
+pub struct Boundary;
+
+impl Default for Boundary {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Boundary {
+    /// Minimum required size for initial process memory.
+    ///
+    /// Need at least 9 dwords of initial stack space for CRT 0:
+    ///
+    /// - 4 dwords for initial upcall arguments
+    /// - 1 dword for initial upcall return address (although this will be zero for init_fn)
+    /// - 4 dwords of scratch space for invoking memop syscalls
+    const MIN_APP_BRK: u32 = 9 * 4;
```

Comment:
I wonder if this should be `9 * core::mem::size_of::<usize>()` instead of a hard-coded 4, so it would more likely not require changes for 64 bit as well in the future?